src/Bathymetry.jl

```
@@ -287,6 +286,10 @@ function remove_minor_basins!(Z, keep_major_basins)
         throw(ArgumentError("`keep_major_basins` must be a finite number!"))
     end
 
+    if keep_major_basins < 1
+        @warn "keep_major_basins is smaller than 1. This will lead to a completely immersed domain."
```

I think you should throw an error.